tool/tsh/dbcmd.go

```diff
@@ -260,20 +268,47 @@ func (c *cliCommandBuilder) isMySQLBinMariaDBFlavor() (bool, error) {
 }
 
 func (c *cliCommandBuilder) getMongoCommand() *exec.Cmd {
+	// look for `mongosh`
+	hasMongosh := c.isMongoshBinAvailable()
+
+	// Starting with Mongo 4.2 there is an updated set of flags.
+	// We are using them with `mongosh` as otherwise warnings will get displayed.
```

Comment:
I don't think I fully understand that comment. Here we only check if `mongosh` is available in the system, but comment says that we should use a new set of flags with Mongo 4.2.

Reply:
For Mongo 4.2 using deprecated flags (e.g. `--ssl` instead of `--tls`) will cause warnings to be shown. This is expected behaviour. To avoid that we would have to fiddle with parsing of `mongo --version` result which feels like too much work for too little gain. Since we know that `mongosh` supports the updated set of flags we use it in case we are using the new client.